### PR TITLE
Only show icon for well-known chains

### PR DESCRIPTION
**Pull request by tomaka**
This PR makes it so the extension knows what is a well-known chain and what isn't.

This way, we can show the icon of a chain only if it is well-known.

This is important for security. Right now we just do `if icon.get(chainName)`, but given that `chainName` is untrusted user input, anyone can pretend that they're connecting to for example Polkadot while in fact connecting to a chain that isn't Polkadot.

@wirednkod Could you push on this PR and make it so that the `IconWeb3` also shows `?` if `isWellKnown` is `false`? I don't know how to pass values around with React.

### CODE REVIEW

**Review by wirednkod — APPROVED** [review]

**Comment by tomaka** [comment]
Thank you

File: projects/extension/src/components/IconWeb3.tsx [hunk 1]
```diff
@@ -10,10 +11,13 @@ interface Props {
 const hasGlyph = (string: string) =>
   ["kusama", "polkadot", "westend", "rococo"].indexOf(string) > -1
```
Comment (gluneau) [inline projects/extension/src/components/IconWeb3.tsx:12]: Will I be allowed to parachains like Astar, Shiden & GM?

My previous PR was reverted.
https://github.com/paritytech/substrate-connect/pull/1315

Comment (tomaka) [reply projects/extension/src/components/IconWeb3.tsx:12]: No, only these 4 relay chains have an icon because we have explicitly integrated them into substrate-connect.
We don't want to curate a list of chains like PolkadotJS does.

The reason why these 4 relay chains are hardcoded in the extension is because, specifically for relay chains, it makes syncing them much faster. And because this is a Parity product, we're also skewed towards putting the Parity chains in the code. This optimization, however, is irrelevant for parachains. Hardcoding parachains in the code brings absolutely no technical benefit.
So basically it's these 4 chains and no other.
